### ISSUE TITLE
ERROR: column "encrypted" does not exist

### ISSUE THREAD

**Issue by mkarpik79** [waa/baculabackupreport#14]
Hi Bill, 

I'm experiencing below issue with new versions of report. Last one which worked fine was from Apr 19. I'm using Bacula version: 13.0.4 (12 February 2024) x86_64-pc-linux-gnu debian 12.5

Is that related to old DB schema version? What can I do to use your latest report?

```
/etc/bacula/scripts/baculabackupreport.py -C /etc/bacula/scripts/baculabackupreport.ini --dbname bacula --dbuser bacula --dbpass bacula
- Reading configuration overrides from config file '/etc/bacula/scripts/baculabackupreport.ini', section 'DEFAULT' (if exists), and section 'baculabackupreport'

Problem communicating with database 'bacula' while fetching filtered jobs.

Error on line number: 577
ERROR: column "encrypted" does not exist
LINE 1: ...  PriorJobId, AGE(EndTime, StartTime) AS RunTime, Encrypted ...
                                                             ^

pgcode: 42703
```
BTW: I appreciate your work very much!

**Comment by waa** [comment]
Yeah, it seems that the encrypted column first appeared in Bacula Enterprise 16.x (currently 16.0.15), which is tracked by community version 15.x (currently 15.0.2).

At this point, you really only have two choices:

- Upgrade your community version to 15.0.2

- clone the baculabackupreport repository, and then checkout commit `9c456ac2e040b928742f06b72b8de942ac893aaa`, which was the last commit before the encrypted feature was added.

Hope this helps!

Bill

**Comment by DevangRadadiya** [comment]
Fixed issue with Bacula report script in version 13.0.4 due to missing "encrypted" column by removing references to "Encrypted" from SQL query and commenting out the line that used "encrypted" in the script (Line 1641 and Line 1752). Changes include:

**1. Line 1641: Removed Encrypted from the SQL query:**

      PriorJobId, AGE(EndTime, StartTime) AS RunTime

**2. Line 1752: Commented out the line that referenced encrypted:**

         # enc_jobids = {r['jobid']: r['encrypted'] for r in filteredjobsrows if r['type'] in ('B', 'C') and r['jobstatus'] in ('T', 'e') and r['encrypted'] != 0}

**Comment by waa** [comment]
Hello Devang,

Yes, I guess this could qualify as a bullet point #3 in my reply on Jul 31, 2024. :)

But, this means that every time I update the repository and you pull it, then you will need to make those same edits.

I am really of the opinion that the best option is to upgrade to Bacula 15.0.2. :)

Another option I just thought of would be to just alter the database and add the encrypted column to the Job table. This can't really hurt anything.

The SQL query to do this would simply be:
----8<----
ALTER TABLE Job ADD COLUMN Encrypted         int         default 0;
----8<----

That line was just swiped from the 15.0.2 `update_postgresql_tables` script.


Hope this helps!
Bill

**Comment by waa** [comment]
I'll just close this issue as I think all possible fixes have been discussed.
